client/modules/workspace/src/SelectModal/SelectModal.tsx

```diff
@@ -209,78 +294,109 @@ export const SelectModal: React.FC<{
       ),
-    [navCallback, selectedApi, selectedSystem, selectedPath]
+    [navCallback, selectedApi, selectedSystem, selectedPath, selectCallback]
   );
 
   return (
-    <>
-      {React.createElement(children, { onClick: showModal })}
-      <Modal
-        open={isModalOpen}
-        onCancel={handleClose}
-        width="40%"
-        title={<h2>SELECT FILE</h2>}
-        footer={null}
-      >
-        <article className={styles.modalContent}>
-          <div className={styles.modalPanel}>
-            <Select
-              options={systemOptions}
-              style={{ marginBottom: '12px' }}
-              virtual={false}
-              value={dropdownValue}
-              onChange={(newValue) => dropdownCallback(newValue)}
-            />
-            <section className={styles.filesSection}>
-              {!showProjects && (
-                <>
-                  <BaseFileListingBreadcrumb
-                    api={selectedApi}
-                    system={selectedSystem}
-                    path={decodeURIComponent(selectedPath)}
-                    systemRootAlias={selection.projectId}
-                    initialBreadcrumbs={
-                      selectedSystem.startsWith('project-')
-                        ? [{ title: 'My Projects', path: 'PROJECT_LISTING' }]
-                        : []
-                    }
-                    itemRender={(item) => {
-                      return (
-                        <Button
-                          type="link"
-                          onClick={() => navCallback(item.path ?? '')}
-                        >
-                          {item.title}
-                        </Button>
-                      );
-                    }}
-                  />
-                  <div className={styles.filesTableContainer}>
-                    <FileListingTable
-                      api={selectedApi}
-                      system={selectedSystem}
-                      path={selectedPath}
-                      columns={FileColumns}
-                      rowSelection={undefined}
-                      scheme={scheme}
-                      scroll={undefined}
-                    />
-                  </div>
-                </>
-              )}
-              {showProjects && (
-                <div className={styles.destFilesTableContainer}>
-                  <SelectModalProjectListing
-                    onSelect={(uuid, projectId) =>
-                      onProjectSelect(uuid, projectId)
-                    }
-                  />
-                </div>
-              )}
-            </section>
+    <Modal
+      open={isModalOpen}
+      onCancel={handleClose}
+      width="40%"
+      footer={null}
+      styles={modalStyles}
+      title={<span className={styles.modalTitle}>Select</span>}
+    >
+      <article className={styles.modalContent}>
+        <div className={styles.modalPanel}>
+          <div className={styles.dataFilesModalColHeader}>Select Input</div>
```

Comment:
Perhaps this should be `Select ${input.name}`, i.e. "Select Case Directory"

Reply:
done